src/lib/redux/customerDataSlice.ts

```diff
@@ -1,9 +1,58 @@
 /* eslint-disable no-unused-vars */
 /* eslint-disable no-param-reassign */
 import { createEntityAdapter, createSlice, EntityState } from '@reduxjs/toolkit';
-import { CustomerRecord, SiteId } from '../airtable/interface';
+import { CustomerRecord, MeterReadingRecord, PaymentRecord, SiteId } from '../airtable/interface';
 import { setCurrSite } from './siteDataSlice';
 import { RootState } from './store';
+import moment from 'moment';
+
+const customersAdapter = createEntityAdapter<CustomerRecord>();
+const paymentsAdapter = createEntityAdapter<PaymentRecord>({
+    // Sort by descending timestamp
+    sortComparer: (a, b) => moment(b.date).diff(a.date),
+});
+const meterReadingsAdapter = createEntityAdapter<MeterReadingRecord>({
+    // Sort by descending timestamp
+    sortComparer: (a, b) => moment(b.date).diff(a.date),
+});
+
+// Returns customers in the context of the current site
+export const {
+    selectEntities: selectAllCustomers,
+    selectAll: selectAllCustomersArray,
+    selectById: selectCustomerById,
+    selectIds: selectCustomerIds
+} = customersAdapter.getSelectors(
+    (state: RootState) => state.customerData.sitesCustomers[state.siteData.currentSite.id].customers);
+
+export const {
+    selectEntities: selectAllPayments,
+    selectAll: selectAllPaymentsArray,
+    selectById: selectPaymentById,
+    selectIds: selectPaymentIds
+} = paymentsAdapter.getSelectors(
+    (state: RootState) => state.customerData.sitesCustomers[state.siteData.currentSite.id].payments);
+
+export const {
+    selectEntities: selectAllMeterReadings,
+    selectAll: selectAllMeterReadingsArray,
+    selectById: selectMeterReadingById,
+    selectIds: selectMeterReadingIds
+} = meterReadingsAdapter.getSelectors(
+    (state: RootState) => state.customerData.sitesCustomers[state.siteData.currentSite.id].meterReadings);
+
+interface CustomerData {
```

Comment:
Maybe rename to more explicitly mention that this is customer data related to a specific site. I called it `SiteInventoryData` in inventory to delineate the difference.

Reply:
Renamed to `SiteCustomerData` to match, thanks!